app/api/users.py

```
@@ -16,23 +15,12 @@ async def create_onboarding(
     """
     return await user_service.create_complete_onboarding(onboarding_data)
 
-@router.get("/onboarding")
+@router.get("/{user_id}/onboarding")
 async def get_onboarding_status(
     x_user_id: str = Header(..., alias="X-User-Id"),
     user_service: UserService = Depends(get_user_service)
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)